website/src/theme/BlogPostItem/index.js

```diff
@@ -101,6 +101,38 @@ function BlogPostItem(props) {
     );
   };
 
+  // dbt custom - send blog context to datalayer to send to snowplow
+  useEffect(() => {
+    window.dataLayer = window.dataLayer || [];
+
+    // Only send to datalayer if blog post page
+    let blogContext = {
+      blogAuthor: '',
+      blogCategory: '',
+      blogDate: (formattedDate && isBlogPostPage) ? formattedDate : undefined
+    }
+
+    if(authors && authors.length > 0 && isBlogPostPage) {
+      authors.map((author, i) => {
+        blogContext.blogAuthor += 
+          `${author.name}${i !== authors.length - 1 ? ', ' : ''}`
+      })
+    } else {
+      blogContext.blogAuthor = undefined
```

Comment:
I'm assuming it's okay that these values are going into GTM as "undefined?" I'm not sure how GTM treats those values.

Reply:
Yes that sets the variables to `undefined` in GTM, which is intended. This component lives on other pages such as the blog index page. If on any page other than a blog post page, we want to make sure we update the data layer variables to undefined, otherwise GTM will use the value which was last pushed.

For example, if someone views a blog post page, the author, category and date is sent to GTM as variables. If we didn't update these variables to `undefined` when clicking a different page, when the visitor clicks a different page it would still pass in the original values from the blog post page into Snowplow.

This ensures we're updating the variables on every page load, that way those variables will only have values if viewing a blog post page, otherwise it's passed in as `undefined` for all other pages.

Reply:
Yeah it would go through like the screenshot below. I followed up in the asana task to verify if that'll work or if we should make any adjustments to improve it on the data side.
![image](https://user-images.githubusercontent.com/13647110/199537581-980acd0c-d4fc-43f5-a91d-93f9dfbbe1c0.png)